### PR TITLE
bpo-43224: Add tests for TypeVarTuple substitution in Annotated

### PR DESCRIPTION
**Pull request by mrahtz**
At some point I guess we should add tests for the native `tuple` too, once https://github.com/python/cpython/pull/31828 or similar is in.

@JelleZijlstra Does this assuage the concerns you mentioned about `Annotated` in https://github.com/python/cpython/pull/31021#pullrequestreview-890627941?

@@GBeauregard too, since Jelle also mentioned you in that thread - is there anything else we should test for here?

https://bugs.python.org/issue43224

### CODE REVIEW

File: Lib/typing.py [hunk 1]
```diff
@@ -2046,6 +2057,9 @@ def __class_getitem__(cls, params):
             raise TypeError("Annotated[...] should be used "
                             "with at least two arguments (a type and an "
                             "annotation).")
+        if _is_unpacked_typevartuple(params[0]):
```
Comment (JelleZijlstra) [inline Lib/typing.py:2107]: I don't think we need to explicitly disallow this at runtime.

**Review by JelleZijlstra — COMMENTED** [review]
Thanks! The tests are good but it's a bit unfortunate if we can't use Annotated around a TypeVarTuple at all.

File: Lib/typing.py (hunk 1 above)
Comment (mrahtz) [reply Lib/typing.py:2107]: Hmm, but then how should we handle situations like this?

```python
A = Annotated[Unpack[Ts], "foo"]
B = A[int, str]
```

`Annotated[int, str, "foo"]` wouldn't be valid. We could do some kind of automatic tuple wrapping, `Annotated[tuple[int, str], "foo"`, but that would be inconsistent with how we handle `TypeVarTuple` substitution elsewhere.

Comment (JelleZijlstra) [reply Lib/typing.py:2107]: Good point, and sorry for not responding to this earlier.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**